library/src/main/java/com/mux/player/internal/cache/CacheDatastore.kt

```diff
@@ -162,6 +182,34 @@ internal class CacheDatastore(val context: Context) : Closeable {
    */
   fun indexDbDir(): File = File(context.filesDirNoBackupCompat, CacheConstants.CACHE_BASE_DIR)
 
+  /**
+   * Generates a URL-safe cache key for a given URL. Delegates to [generateCacheKey] but encodes it
+   * into something else
+   */
+  fun safeCacheKey(url: URL): String = Base64.encodeToString(
```

Comment:
Good call to separate the logic for constructing a safe cache key out, will do likewise